### ISSUE TITLE
Issues with package extensions

### ISSUE THREAD

**Issue by mateuszbaran** [JuliaDocs/Documenter.jl#2124]
I'm working on introducing package extensions to one of my packages: https://github.com/JuliaManifolds/Manifolds.jl/pull/607 . I've encountered some issues that would be most easily explained by Documenter not loading (or not noticing) the code from extensions.

First, docstrings written in a package extensions are not rendered even if the extension is loaded in the Julia session from which Documenter is triggered.

Second, `@example` block don't seem to load package extensions when their code is executed which leads to errors that don't happen when the same code is just executed in a new Julia REPL.

I'm not 100% sure if these are Documenter.jl bugs but maybe someone has experienced similar issues?

**Comment by mortenpi** [comment]
Do you happen to have an MWE where I could try this out?

**Comment by mateuszbaran** [comment]
I've found a workaround: passing the extension module to `makedocs`: https://github.com/JuliaManifolds/Manifolds.jl/pull/607/commits/4203a96c3684360fed5b40a41bff615eac0ede06#diff-4aae2d1c783cade58bd2cb13748da956e568b7f2aed5fafd9e2a46fb97daf613L73-R77 . Maybe it would be a good idea to document this pattern somewhere?

**Comment by mortenpi** [comment]
Yeah, that's what I presume you need to do, since they're their own modules. Having some notes about it in the Documenter manual would be useful I think.

**Comment by oschulz** [comment]
Thanks for the workaround @mateuszbaran !

**Comment by rafaqz** [comment]
Thanks @mateuszbaran . 

It would be good to automate this in Documenter too. I assumed that loading the packages in the make.jl file would make the extension docs available. It seems more natural to do that than to explicitly get the extensions like that.

**Comment by codetalker7** [comment]
> I've found a workaround: passing the extension module to `makedocs`: [JuliaManifolds/Manifolds.jl@4203a96#diff-4aae2d1c783cade58bd2cb13748da956e568b7f2aed5fafd9e2a46fb97daf613L73-R77](https://github.com/JuliaManifolds/Manifolds.jl/commit/4203a96c3684360fed5b40a41bff615eac0ede06#diff-4aae2d1c783cade58bd2cb13748da956e568b7f2aed5fafd9e2a46fb97daf613L73-R77) . Maybe it would be a good idea to document this pattern somewhere?

Hi. Related to this issue: I had a few docstrings in a package extension which I wanted to include in the main docs, _without_ having to load the trigger package for the extension (because, the trigger package was a GPU package, and my system doesn't have support for that particular GPU). It seems that the trick mentioned here doesn't work for this case. It would be great if there was some functionality in package extensions to fetch only docstrings from extensions (without having to install trigger packages or run extension code).

**Comment by kbarros** [comment]
In my main package I create a stub:
```julia
function extension_function end
```

and in my `ExtensionPackage` I define the method and attach a doc string:
```julia
"""Long doc string"""
MainPackage.extension_function() = println("hello world")
```

Even if I inform Documenter about the presence of `ExtensionPackage`, it seems that Documenter is unable pick up the doc string that is defined by `ExtensionPackage` into `MainPackage`. Is there a trick for transferring the doc string across packages?

**Comment by BambOoxX** [comment]
Is there a way to document internal functions in extensions ? Although they are not accessible in the main package, a documentation may be a relevant source of information...
